fakeredis/_aioredis2.py

```diff
@@ -0,0 +1,170 @@
+import asyncio
+from typing import Union
+
+import aioredis
+
+from . import _async, _server
+
+
+class FakeSocket(_async.AsyncFakeSocket):
+    _connection_error_class = aioredis.ConnectionError
+
+    def _decode_error(self, error):
+        return aioredis.connection.BaseParser(1).parse_error(error.value)
+
+
+class FakeReader:
+    pass
```

Comment:
Out of curiosity, why don't you need a Reader with aioredis 2?

Reply:
fakeredis is a bit asymmetric (and one day I might try to address that): commands get serialised by aioredis/redis-py and parsed by fakeredis, but responses are kept as structured Python objects and the response parsing is bypassed.